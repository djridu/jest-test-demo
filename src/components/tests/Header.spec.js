import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { render, fireEvent, waitForElement } from 'react-testing-library';
import Header from '../Header';
import TodoTextInput from '../TodoTextInput';

const setup = () => {
    const props = {
        addTodo: jest.fn(),
    };

    const renderer = new ShallowRenderer();
    renderer.render(<Header {...props} />);
    const output = renderer.getRenderOutput();

    return {
        props,
        output,
        renderer,
    };
};

describe('components', () => {
    describe('Header', () => {
        it('should render and match the snapshot', () => {
            const output = setup();
            expect(output).toMatchSnapshot();
        });

        it('should render correctly', () => {
            const { output } = setup();
            expect(output.type).toBe('header');
            expect(output.props.className).toBe('header');

            const [h1, input] = output.props.children;
            expect(h1.type).toBe('h1');
            expect(h1.props.children).toBe('todos');
            expect(input.type).toBe(TodoTextInput);
            expect(input.props.newTodo).toBe(true);
            expect(input.props.placeholder).toBe('What needs to be done?');
        });

        it('should call addTodo if length of text is greater than 0', () => {
            const { output, props } = setup();
            const input = output.props.children[1];
            input.props.onSave('');
            expect(props.addTodo).not.toBeCalled();
            input.props.onSave('Use Redux');
            expect(props.addTodo).toBeCalled();
        });

        it('should call addTodo if length of text is greater than 0 - v2', () => {
            const { output, props } = setup();
            const input = output.props.children[1];
            input.props.onSave('');
            expect(props.addTodo).not.toBeCalled();
            input.props.onSave('Use Redux');
            expect(props.addTodo).toBeCalled();
        });
    });
});
