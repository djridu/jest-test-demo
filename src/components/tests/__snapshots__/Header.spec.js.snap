// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components Header should render and match the snapshot 1`] = `
Object {
  "output": <header
    className="header"
  >
    <h1>
      todos
    </h1>
    <TodoTextInput
      newTodo={true}
      onSave={[Function]}
      placeholder="What needs to be done?"
    />
  </header>,
  "props": Object {
    "addTodo": [MockFunction],
  },
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_currentlyRenderingComponent": null,
    "_didScheduleRenderPhaseUpdate": false,
    "_dispatcher": Object {
      "readContext": [Function],
      "useCallback": [Function],
      "useContext": [Function],
      "useDebugValue": [Function],
      "useEffect": [Function],
      "useImperativeHandle": [Function],
      "useLayoutEffect": [Function],
      "useMemo": [Function],
      "useReducer": [Function],
      "useRef": [Function],
      "useState": [Function],
    },
    "_element": <Header
      addTodo={[MockFunction]}
    />,
    "_firstWorkInProgressHook": null,
    "_forcedUpdate": false,
    "_instance": null,
    "_isReRender": false,
    "_newState": null,
    "_numberOfReRenders": 0,
    "_previousComponentIdentity": [Function],
    "_renderPhaseUpdates": null,
    "_rendered": <header
      className="header"
    >
      <h1>
        todos
      </h1>
      <TodoTextInput
        newTodo={true}
        onSave={[Function]}
        placeholder="What needs to be done?"
      />
    </header>,
    "_rendering": false,
    "_updater": Updater {
      "_callbacks": Array [],
      "_renderer": [Circular],
    },
    "_workInProgressHook": null,
  },
}
`;
